homeassistant/components/swiss_public_transport/helper.py

```diff
@@ -57,3 +69,24 @@ def unique_id_from_config(config: MappingProxyType[str, Any] | dict[str, Any]) -
             else ""
         )
     )
+
+
+def update_stats(stats: dict[str, Stats], success: bool = True) -> None:
+    """Update the stats and record every call for the last couple of days."""
+    today = dt_util.now().date()
+    today_key = today.isoformat()
+    keep_dates = [
+        (today - timedelta(days=i)).isoformat() for i in range(DIAGNOSE_STATS_MAX_AGE)
+    ]
+    if today_key not in stats:
+        stats[today_key] = Stats()
+    for key in [key for key in stats if key not in keep_dates]:
+        del stats[key]
+
+    time_value = dt_util.now().time().isoformat()
+    stats[today_key].count += 1
+    if success:
+        stats[today_key].timestamps_success.append(time_value)
+    else:
+        stats[today_key].errors += 1
+        stats[today_key].timestamps_error.append(time_value)
```

Comment:
I am wondering how useful this data is, as it requires you to have a running HA instance, and the moment you reload the integration or restart HA, your data is gone. What kind of data do you need from history to be able to diagnose?